util/predicates/generic_predicates.go

```diff
@@ -172,6 +178,25 @@ func ResourceHasFilterLabel(scheme *runtime.Scheme, logger logr.Logger, labelVal
 	}
 }
 
+// ResourceHasFilterExcludeNamespaces returns a predicate that returns true only if the provided resource contains
+// a matching namespace.
+func ResourceHasFilterExcludeNamespaces(scheme *runtime.Scheme, logger logr.Logger, excludedNamespaces []string) predicate.Funcs {
+	return predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return processIfNamespaceExcluded(scheme, logger.WithValues("predicate", "ResourceHasFilterExcludeNamespaces", "eventType", "update"), e.ObjectNew, excludedNamespaces)
+		},
+		CreateFunc: func(e event.CreateEvent) bool {
+			return processIfNamespaceExcluded(scheme, logger.WithValues("predicate", "ResourceHasFilterExcludeNamespaces", "eventType", "create"), e.Object, excludedNamespaces)
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return processIfNamespaceExcluded(scheme, logger.WithValues("predicate", "ResourceHasFilterExcludeNamespaces", "eventType", "delete"), e.Object, excludedNamespaces)
+		},
+		GenericFunc: func(e event.GenericEvent) bool {
+			return processIfNamespaceExcluded(scheme, logger.WithValues("predicate", "ResourceHasFilterExcludeNamespaces", "eventType", "generic"), e.Object, excludedNamespaces)
+		},
+	}
+}
+
```

Comment:
I don't think this is the right approach to solving this issue. The suggestion in #11193 uses the FieldSelector concept of an API request to filter resources at the API server level, rather than the controller level.

Given the scale potential here, I think we want to follow that suggestion and move the filtering to the API server where possible.

I'm not sure if that approach also has RBAC advantages, perhaps we don't need permissions on the excluded namespace? But that could also be an advantage of the approach as suggested by Alberto

Reply:
Thanks, @JoelSpeed! Please take a look at the new approach in this PR.